apps/io-profile-async/src/repositories/session-notifications.ts

```diff
@@ -66,8 +71,30 @@ export const updateNotificationEvents: (
     notificationEvents
   });
 
+// TODO: add tests
+const updateExpiredSessionNotificationFlag: (
+  actualRecord: RetrievedSessionNotifications,
+  flagNewValue: boolean
+) => RTE.ReaderTaskEither<Dependencies, CosmosErrors, void> = (
+  actualRecord: RetrievedSessionNotifications,
+  flagNewValue: boolean
+) => (deps: Dependencies) =>
+  pipe(
+    deps.sessionNotificationsModel.patch(
+      [actualRecord.id, actualRecord.expiredAt],
+      {
+        notificationEvents: {
+          ...actualRecord.notificationEvents,
+          EXPIRED_SESSION: flagNewValue
+        }
```

Comment:
`patch` will only affect declared properties, so there is no need to spread all other notificationEvents.

See [implementation](https://github.com/pagopa/io-functions-commons/blob/1cf04d9ce5a96e7f7bbe8667cf8c818530c0b28c/src/utils/cosmosdb_model.ts#L252) with operation type `add` and related [documentation](https://learn.microsoft.com/en-us/azure/cosmos-db/partial-document-update#supported-operations)

```suggestion
        notificationEvents: {
          EXPIRED_SESSION: flagNewValue
        }
```